ccp/src/utility_thread/thread.rs

```diff
@@ -101,24 +101,39 @@ impl UtilityThreadImpl {
     }
 
     pub(crate) async fn utility_closure(mut self, core_ids: Vec<LogicalCoreId>) {
-        use crossterm::event::EventStream;
-        use futures::FutureExt;
-        use futures::StreamExt;
-
         if !cpu_utils::pinning::pin_current_thread_to_cpuset(core_ids.iter().cloned()) {
             log::error!("failed to pin to {core_ids:?} cores");
         }
 
         let mut cum_hashrate_ticker =
             time::interval(Duration::from_secs(CUMULATIVE_HASHRATE_UPDATE_INTERVAL));
 
-        let mut terminal_event_reader = EventStream::new();
+        #[cfg(feature = "crossterm")]
+        let mut terminal_event_reader = crossterm::event::EventStream::new();
+
+        #[cfg(feature = "crossterm")]
+        {
+            use futures::FutureExt;
+            use futures::StreamExt;
+
+            loop {
+                tokio::select! {
+                Some(message) = self.to_utility.recv() => self.handle_to_utility_message(message).await,
+                _ = cum_hashrate_ticker.tick() => self.handle_cum_hashrate_tick().await,
+                maybe_event = terminal_event_reader.next().fuse() => self.handle_terminal_event(maybe_event).await,
```

Comment:
Duplicate code is hell to maintain.
I may suggest defining
```rust
        #[cfg(not(feature = "crossterm"))]
        let mut terminal_event_reader = futures::stream::pending();
```
and some dummy implementation of `handle_terminal_event`.

Reply:
ofc, this PR is just for testing purposes